audobject/core/utils.py

```diff
@@ -1,22 +1,62 @@
 import importlib
 import inspect
+import subprocess
+import sys
+import types
 import typing
 import warnings
 
+import audeer
+
 from audobject.core.config import config
 from audobject.core import define
 
 
-def get_class(key: str) -> (type, str, str):
-    r"""Load class module."""
+def get_class(
+        key: str,
+        auto_install: bool,
+) -> (type, str, str):
+    r"""Load class."""
     if key.startswith(define.OBJECT_TAG):
         key = key[len(define.OBJECT_TAG):]
-    module_name, class_name, version = split_key(key)
+    package_name, module_name, class_name, version = split_key(key)
+    module = get_module(
+        package_name,
+        module_name,
+        version,
+        auto_install,
+    )
     installed_version = get_version(module_name)
-    module = importlib.import_module(module_name)
     return getattr(module, class_name), version, installed_version
 
 
+def get_module(
+        package_name: str,
+        module_name: str,
+        version: typing.Optional[str],
+        auto_install: bool,
+) -> types.ModuleType:
+    r"""Load module."""
+    try:
+        module = importlib.import_module(module_name)
+    except ModuleNotFoundError as ex:
+        if auto_install:
+            audeer.install_package(
+                package_name,
+                version=version,
+            )
```

Comment:
`audeer.install_package()` will raise an error if a different version is already installed, or do you use `>=` or something here?

Reply:
We only call `install_package()` if a `ModuleNotFoundError` is raised, i.e. no version of the package is installed yet.